site/src/pages/release-notes/1.29.0.mdx

```diff
@@ -0,0 +1,257 @@
+- <type://SamlSingleSignOnHandler#loginFailed(ServiceRequestContext,AggregatedHttpRequest,MessageContext,Throwable)>
+  can now access the <type://MessageContext> for error handling if available. #5401 #5622
+- The copied cURL command from a doc service, is now quoted correctly. #5566
+
+## 🛠️ Bug fixes
+
+- <type://HealthCheckedEndpointGroup> no longer raises an <type://EndpointSelectionTimeoutException> when
+  a server is stopped by the JVM shutdown hook. #5718
+- Fixed a bug where <type://RetryingClient> gets to deadlock when <type://OAuth2Client> fails to obtain
+  an access token. #5715
+- Armeria client always uses HTTP/2 connection preface for `h2c`, regardless of
+  the value of <type://ClientFactoryOptions#useHttp2Preface()>. #5706
+- <type://ArmeriaServerConfigurator> now properly overrides the properties set by <type://ArmeriaSettings>. #5009 #5692
+- Armeria server now returns a 408 status if a service didn't receive the request fully and
+  the request times out. #5579 #5680
+- A `NullPointerException` isn't raised anymore when running Armeria with a self-signed certificate. #5669
+  It happened when the following conditions are all met:
+  - DEBUG-level logging of JDK security event is enabled.
+  - A user has no Bouncy Castle dependency.
+- A `DecompressionException` isn't ignored anymore in a <type://StreamDecoder>. #5177 #5620
+- <type://@Description> annotations at the super class or interface are now used for the parameter description. #5195 #5562
+- The specfied `ObjectMapper` to <type://WebClientRequestPreparation#asJson(Class,ObjectMapper)> isn't
+  ignored anymore. #5454 #5512
+- <type://GrpcExceptionHandlerFunction> is now applied to convert a gRPC cancellations. #5329 #5398
+- In Spring integration, the default <type://ClientFactory> is now gracefully closed after the server shuts down. #5742
+- A connection is not reused anymore after a <type://WriteTimeoutException> is raised. #5738
+
+## 🏚️ Deprecations
+
+- <type://PrometheusMeterRegistries> and <type://PrometheusExpositionService> are deprecated. #5698
+  - Use the same classes in the `armeria-prometheus1` module.
+- <type://HttpResult> is deprecated. Use <type://ResponseEntity> instead. #4910 #5586
+- <type://GraphqlConfigurator> and various setter methods for building a `GraphQL` instance are deprecated.
+  - Use <type://GraphqlServiceBuilder#graphql(GraphQL)> instead. #5269
+
+## ☢️ Breaking changes
+
+- We updated Micrometer to 1.13.0 that has breaking changes in its Prometheus support. #5698
+  - If you want to keep the old behavior that uses Prometheus Java client 0.x,
+    use the `io.micrometer:micrometer-registry-prometheus-simpleclient:1.13.0` module.
+  - If you want to use Prometheus Java client 1.x, add `com.linecorp.armeria:armeria-prometheus1` module.
+  - More details can be found in the [migration guide](https://github.com/micrometer-metrics/micrometer/wiki/1.13-Migration-Guide).
```

Comment:
nit: Because it isn't our migration guide.
```suggestion
  - More details can be found in the [Micrometer migration guide](https://github.com/micrometer-metrics/micrometer/wiki/1.13-Migration-Guide).
```